website/content/docs/setup/vault-configuration/index.md

```diff
@@ -0,0 +1,124 @@
+---
+title: Vault Configuration
+linktitle: Vault Configuration
+weight: 30
+---
+
+The {{< name >}} uses [HashiCorp Vault](https://www.vaultproject.io/) to provide [Modules](../../overview/modules) running in the remote clusters with the dataset credentials to execute their tasks.
+
+This page contains two sections:
+
+1. [This section](#configure-vault-kubernetes-auth-method-in-each-cluster) contains the required steps to enable the modules to authenticate to Vault in order for them to retrieve the dataset credentials.
+
+2. [This section](#add-vault-secret-plugin) describes how to add a new custom [Vault secret plugin](https://www.vaultproject.io/docs/plugin) to retrieve the dataset credentials from where they are stored.
+
+
+## Configure Vault Kubernetes auth method in each cluster
+
+Module authentication is done by configuring Vault to use [Kubernetes auth method](https://www.vaultproject.io/docs/auth/kubernetes) in each cluster. Using this method the modules can authenticate to Vault by providing their service account token. Behind the scenes Vault authenticates the token by submitting TokenReview request to the API server of the kubernetes cluster where the module is running. 
+
+Some of the steps described below are not specific to the {{< name >}} project but rather are Vault specific and can be found in Vault related online tutorials.
+
+It is assumed that Vault is deployed in the coordinator cluster in `m4d-system` namespace prior to executing the steps.
+
+1. Setup Ingress to Vault in the coordinator cluster for multi-cluster setup:
+In multi-cluster setup a kubernetes Ingress should be configured in the coordinator cluster to enable communication to Vault from remote clusters.
+The ingress should be deployed in the `m4d-system` namespace where Vault is deployed.
+
+2. Add Vault address to `m4d-config`:
+The address of Vault should be added to the kubernetes ConfigMap called `m4d-config` in `m4d-system` namespace in the coordinator cluster as follows:
+```bash
+VAULT_ADDRESS: <Vault address>
+```
+In a multi-cluster setup the Vault address should be the Ingress address described above while in single cluster it should set to: `http://vault.m4d-system:8200/`.
+
+**Pre-steps for configuring Vault:**
```

Comment:
please add on which systems/version this was tested on.

Reply:
OK.